std/convert/markdown.ts

```diff
@@ -0,0 +1,1738 @@
+    // but we might need to add _that_
+    // https://tools.ietf.org/html/rfc3986#section-3
+    if (/^[^:]+:\/*[^/]*$/.test(base)) {
+      baseUrls[` ${base}`] = base + "/";
+    } else {
+      baseUrls[` ${base}`] = rtrim(base, "/", true);
+    }
+  }
+  base = baseUrls[` ${base}`];
+  const relativeBase = base.indexOf(":") === -1;
+
+  if (href.slice(0, 2) === "//") {
+    if (relativeBase) {
+      return href;
+    }
+    return base.replace(/^([^:]+:)[\s\S]*$/, "$1") + href;
+  } else if (href.charAt(0) === "/") {
+    if (relativeBase) {
+      return href;
+    }
+    return base.replace(/^([^:]+:\/*[^/]*)[\s\S]*$/, "$1") + href;
+  } else {
+    return `${base}${href}`;
+  }
+}
+
+function splitCells(tableRow: string, count?: number): string[] {
+  // ensure that every cell-delimiting pipe has a space
+  // before it to distinguish it from an escaped pipe
+  const row = tableRow.replace(/\|/g, (_match, offset, str) => {
+    let escaped = false;
+    let curr = offset;
+    while (--curr >= 0 && str[curr] === "\\") escaped = !escaped;
+    if (escaped) {
+      // odd number of slashes means | is escaped
+      // so we leave it alone
+      return "|";
+    } else {
+      // add space before unescaped |
+      return " |";
+    }
+  });
+  const cells = row.split(/ \|/);
+  let i = 0;
+
+  if (cells.length > count!) {
+    cells.splice(count!);
+  } else {
+    while (cells.length < count!) {
+      cells.push("");
+    }
+  }
+
+  for (; i < cells.length; i++) {
+    // leading or trailing whitespace is ignored per the gfm spec
+    cells[i] = cells[i].trim().replace(/\\\|/g, "|");
+  }
+  return cells;
+}
+
+/** Remove trailing `c`s. Equivalent to `str.replace(/c*$/, "")`.
+ *
+ * `/c*$/` is vulnerable to REDOS.
+ *
+ * @param invert Remove suffix of non-`c` chars instead. Default `false`.
+ */
+function rtrim(str: string, c: string, invert = false): string {
+  if (str.length === 0) {
+    return "";
+  }
+
+  // Length of suffix matching the invert condition.
+  let suffLen = 0;
+
+  // Step left until we fail to match the invert condition.
+  while (suffLen < str.length) {
+    const currChar = str.charAt(str.length - suffLen - 1);
+    if (currChar === c && !invert) {
+      suffLen++;
+    } else if (currChar !== c && invert) {
+      suffLen++;
+    } else {
+      break;
+    }
+  }
+
+  return str.substr(0, str.length - suffLen);
+}
+
+function findClosingBracket(str: string, b: string): number {
+  if (str.indexOf(b[1]) === -1) {
+    return -1;
+  }
+  let level = 0;
+  for (let i = 0; i < str.length; i++) {
+    if (str[i] === "\\") {
+      i++;
+    } else if (str[i] === b[0]) {
+      level++;
+    } else if (str[i] === b[1]) {
+      level--;
+      if (level < 0) {
+        return i;
+      }
+    }
+  }
+  return -1;
+}
+
+/**
+ * Core function which takes an input of markdown and returns HTML.
+ */
+export function parse(
+  src: string,
+  opt?: Partial<MarkdownOptions>
+): string | void {
```

Comment:
This should be `renderToHtml()` or something...

There are also a lot of unneeded exports. Can we put all of this away in a `marked_es` subdirectory and export selectively under aliases? Like with [`std/encoding/yaml.ts`](https://github.com/denoland/deno/blob/f07ffe89806392698fb8052691b2826f82f15bbe/std/encoding/yaml.ts).

Reply:
The exports match the library that it was adapted from, including the naming of this function.

Reply:
Yeah, that's why I suggested giving it the vendored code treatment and abstracting over it.

Reply:
I guess there's no need for this extra layer if it goes through a `mod.ts` anyway :p